### PR TITLE
Assignment2 - Sergei Ashikhmin & Timothy Nelson

### PR DESCRIPTION
**Pull request by SergeiAshikhmin**
Sergei - @SergeiAshikhmin 
Timothy - @timmytaters 

Azure link: https://lemon-flower-0c0aec71e.5.azurestaticapps.net/

### CODE REVIEW

**Comment by HadiM2** [comment]
Reviewing

File: Assignment2/tests/game.test.ts
Comment (HadiM2) [inline Assignment2/tests/game.test.ts:1]: make sure to add a test for valid words method

File: Assignment2/scripts/word.ts
Comment (HadiM2) [inline Assignment2/scripts/word.ts:1]: ✅

File: Assignment2/pages/about.vue
Comment (HadiM2) [inline Assignment2/pages/about.vue:1]: ✅

File: Assignment2/components/SettingsDialog.vue
Comment (HadiM2) [inline Assignment2/components/SettingsDialog.vue:1]: ✅

**Review by HadiM2 — COMMENTED** [review]
Create a custom component to show available words ✅

Create/extend a word list component with a method called validWords that returns an array of valid words based on current guesses ✅
Create unit tests for the validWords method ❌
Display the number of valid words based on the entered letters ❌
When the number is clicked, show a dialog with all the available words with scrolling if there are too many to show on the screen. ✅
When a word is clicked, the word should be entered as the guess but not submitted ✅
Add a hover over for the word so it is obvious that it can be clicked. (No CSS) ✅

Convert the sidebar to an App Bar ✅

Name of the app on the left with an icon ✅
Choose an icon for the app (from Material Design or somewhere on the web) ✅
Clicking the name and icon of the app take you to the home page ✅
Hamburger menu on the right ✅
Menu option for an About page ✅
    Create an About page that has a few sentences about this class project and how great Meg is ✅
    Make sure this has a back button ✅

Add a settings dialog ✅

Add a settings selection to the App Bar's menu with a gear icon. This should open this dialog regardless of where you are on the site ✅
Option for dark and light mode ✅
Develop two additional color schemes (with creative names) and allow the user to change to them. Schemes should look good in both light and dark mode ✅
The above two items must be implemented with built in Veutify features ✅

Style the main game page ✅

Make it look like [wordle.com](https://www.nytimes.com/games/wordle/index.html) or the Wordle mobile app ✅
Add gradient to guesses blocks and keys (with CSS) ❌
Add drop shadow to keys (No CSS) ❌
Make the app responsive (No CSS) ✅

Include a link to your Azure App in your PR comments. ✅

Extra Credit

Add an appropriate sound when the UI letter buttons are clicked (5 points) ❌
Add creative styling following good UI/UX practices ❌
    Look at products by Google or some of the vuetify components for good UX/UI ideas
    https://m2.material.io/design/guidelines-over

**Comment by MattA41** [comment]
reviewing

File: Assignment2/components/SettingsDialog.vue [hunk 1]
```diff
@@ -0,0 +1,106 @@
+<template>
```
Comment (MattA41) [inline Assignment2/components/SettingsDialog.vue:1]: I like your use of view components to allow for someone to select themes

File: Assignment2/pages/index.vue [hunk 2]
```diff
@@ -0,0 +1,5853 @@
...
+    game.submitGuess();
+  } else if (event.key == "Backspace") {
+    game.removeLastLetter();
+  } else if (event.key.match(/[A-z]/) && event.key.length === 1) {
+    game.addLetter(event.key.toUpperCase());
+  }
+}
+
+// From: https://github.com/charlesreid1/five-letter-words/blob/master/sgb-words.txt
+var WordList = [
```
Comment (MattA41) [inline Assignment2/pages/index.vue]: You do not have to write out the whole word list you can import it from the .ts document. Having it somewhere else would make it more readable

File: Assignment2/plugins/vuetify.ts [hunk 3]
```diff
@@ -0,0 +1,187 @@
+// import this after install `@mdi/font` package
+import "@mdi/font/css/materialdesignicons.css";
+
+import "vuetify/styles";
+import { createVuetify } from "vuetify";
+
+export default defineNuxtPlugin((app) => {
```
Comment (MattA41) [inline Assignment2/plugins/vuetify.ts:7]: I like the themes you made

**Review by MattA41 — COMMENTED** [review]
Over view: It looks good, you guys did some stuff differently than I did but that does not mean its wrong. 

EWU-CSCD378-2024-Spring

## Assignment 2

The purpose of this assignment is to solidify your learning of:
- Front end component development
- Using Vuetify's components
- Building a new custom component
- Creating new pages

## Instructions
Note: Custom CSS can only be used where specified.

1. Create a custom component to show available words ✅
- Create/extend a word list component with a method called validWords that returns an array of valid words based on current guesses ❌
- Create unit tests for the validWords method ❌
- Display the number of valid words based on the entered letters ❌
- When the number is clicked, show a dialog with all the available words with scrolling if there are too many to show on the screen. ✅
- When a word is clicked, the word should be entered as the guess but not submitted ✅
- Add a hover over for the word so it is obvious that it can be clicked. (No CSS) ✅

2. Convert the sidebar to an App Bar ✅
- Name of the app on the left with an icon ✅
- Choose an icon for the app (from Material Design or somewhere on the web) ✅
- Clicking the name and icon of the app take you to the home page ✅
- Hamburger menu on the right ✅
- Menu option for an About page ✅
  - Create an About page that has a few sentences about this class project and how great Meg is ✅
  - Make sure this has a back button ✅

3. Add a settings dialog ✅
- Add a settings selection to the App Bar's menu with a gear icon. This should open this dialog regardless of where you are on the site ✅
- Option for dark and light mode ✅
- Develop two additional color schemes (with creative names) and allow the user to change to them. Schemes should look good in both light and dark mode ✅
- The above two items must be implemented with built in Veutify features ✅

4. Style the main game page ✅
- Make it look like [wordle.com](https://www.nytimes.com/games/wordle/index.html) or the Wordle mobile app ✅
- Add gradient to guesses blocks and keys (with CSS) ✅
- Add drop shadow to keys (No CSS) ✅
- Make the app responsive (No CSS) ✅

5. Include a link to your Azure App in your PR comments.

## Extra Credit

- Add an appropriate sound when the UI letter buttons are clicked (5 points)
- Add creative styling following good UI/UX practices
  - Look at products by Google or some of the vuetify components for good UX/UI ideas
  - https://m2.material.io/design/guidelines-overview

File: Assignment2/tests/game.test.ts
Comment (SergeiAshikhmin) [reply Assignment2/tests/game.test.ts:1]: Will do :)

File: Assignment2/components/SettingsDialog.vue (hunk 1 above)
Comment (SergeiAshikhmin) [reply Assignment2/components/SettingsDialog.vue:1]: Thank you :)

File: Assignment2/pages/index.vue (hunk 2 above)
Comment (SergeiAshikhmin) [reply Assignment2/pages/index.vue]: Yeah, I know. This feature was in development and we didn't care about clean code at that moment. But thanks for the catch. 😄

File: Assignment2/plugins/vuetify.ts (hunk 3 above)
Comment (SergeiAshikhmin) [reply Assignment2/plugins/vuetify.ts:7]: Thank you

File: Assignment2/tests/game.test.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  expect(updateValidWords(game)).not.toContain("snare");
+  expect(updateValidWords(game)).not.toContain("traps");
+  expect(updateValidWords(game)).not.toContain("apart");
+  expect(updateValidWords(game)).not.toContain("aptly");
+
+  expect(updateValidWords(game)).toContain("hello");
+  expect(updateValidWords(game)).toContain("holds");
+  expect(updateValidWords(game)).toContain("smell");
+  expect(updateValidWords(game)).toContain("messy");
+  expect(updateValidWords(game)).toContain("elves");
```
Comment (A-Tanner) [inline Assignment2/tests/game.test.ts:53]: Extensive!

**Review by A-Tanner — COMMENTED** [review]
1. Create a custom component to show available words ✅
- Create/extend a word list component with a method called validWords that returns an array of valid words based on current guesses ✅
- Create unit tests for the validWords method ✅
- Display the number of valid words based on the entered letters ✅
- When the number is clicked, show a dialog with all the available words with scrolling if there are too many to show on the screen. ✅
- When a word is clicked, the word should be entered as the guess but not submitted ✅
- Add a hover over for the word so it is obvious that it can be clicked. (No CSS) ✅

2. Convert the sidebar to an App Bar ✅
- Name of the app on the left with an icon ✅
- Choose an icon for the app (from Material Design or somewhere on the web) ✅
- Clicking the name and icon of the app take you to the home page ✅
- Hamburger menu on the right ✅
- Menu option for an About page ✅
  - Create an About page that has a few sentences about this class project and how great Meg is ✅
  - Make sure this has a back button ✅

3. Add a settings dialog ✅
- Add a settings selection to the App Bar's menu with a gear icon. This should open this dialog regardless of where you are on the site ✅
- Option for dark and light mode ✅
- Develop two additional color schemes (with creative names) and allow the user to change to them. Schemes should look good in both light and dark mode ✅
- The above two items must be implemented with built in Veutify features ✅

4. Style the main game page ✅
- Make it look like [wordle.com](https://www.nytimes.com/games/wordle/index.html) or the Wordle mobile app ✅
- Add gradient to guesses blocks and keys (with CSS) ✅
- Add drop shadow to keys (No CSS) ✅
- Make the app responsive (No CSS) ✅

5. Include a link to your Azure App in your PR **comments.✅**
## Extra Credit

- Add an appropriate sound when the UI letter buttons are clicked (5 points)
- Add creative styling following good UI/UX practices
  - Look at products by Google or some of the vuetify components for good UX/UI ideas
  - https://m2.material.io/design/guidelines-overview

File: Assignment2/pages/index.vue [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        :key="i"
+        :guess="guess"
+      />
+
+      <div class="my-10">
+        <Keyboard />
+      </div>
+
+      <WordList
+      v-if="game.gameState === GameState.Playing"
```
Comment (A-Tanner) [inline Assignment2/pages/index.vue:35]: Nice attention to detail, it adds a level of polish to the experience

File: Assignment2/pages/about.vue [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+				</div>
+			</v-col>
+		</v-row>
+		<v-row justify="center">
+			<v-col cols="12" md="8">
+				<h1 class="ext-h4 my-4 text-center">
+					About Our Teachers
+				</h1>
+				<div class="text-body-1 mb-3">
+					<strong>Grant</strong> and <strong>Mag</strong> are the instructors for this course. They are both very
```
Comment (A-Tanner) [inline Assignment2/pages/about.vue:43]: Who is this Mag person? 😅

File: Assignment2/tests/game.test.ts (hunk 4 above)
Comment (A-Tanner) [inline Assignment2/tests/game.test.ts:53]: Extensive! While manual tests are sufficient, consider ways to test this logic in more general cases

**Review by A-Tanner — COMMENTED** [review]
1. Create a custom component to show available words ✅
- Create/extend a word list component with a method called validWords that returns an array of valid words based on current guesses ✅
- Create unit tests for the validWords method ✅
- Display the number of valid words based on the entered letters ✅
- When the number is clicked, show a dialog with all the available words with scrolling if there are too many to show on the screen. ✅
- When a word is clicked, the word should be entered as the guess but not submitted ✅
- Add a hover over for the word so it is obvious that it can be clicked. (No CSS) ✅

2. Convert the sidebar to an App Bar ✅
- Name of the app on the left with an icon ✅
- Choose an icon for the app (from Material Design or somewhere on the web) ✅
- Clicking the name and icon of the app take you to the home page ✅
- Hamburger menu on the right ✅
- Menu option for an About page ✅
  - Create an About page that has a few sentences about this class project and how great Meg is ✅
  - Make sure this has a back button ✅

3. Add a settings dialog ✅
- Add a settings selection to the App Bar's menu with a gear icon. This should open this dialog regardless of where you are on the site ✅
- Option for dark and light mode ✅
- Develop two additional color schemes (with creative names) and allow the user to change to them. Schemes should look good in both light and dark mode ✅
- The above two items must be implemented with built in Veutify features ✅

4. Style the main game page ✅
- Make it look like [wordle.com](https://www.nytimes.com/games/wordle/index.html) or the Wordle mobile app ✅
- Add gradient to guesses blocks and keys (with CSS) ✅
- Add drop shadow to keys (No CSS) ✅
- Make the app responsive (No CSS) ✅

5. Include a link to your Azure App in your PR comments.✅
## Extra Credit

- Add an appropriate sound when the UI letter buttons are clicked (5 points)
- Add creative styling following good UI/UX practices
  - Look at products by Google or some of the vuetify components for good UX/UI ideas
  - https://m2.material.io/design/guidelines-overview
  
  Great work you two! Solid implementation and an example of good teamwork. I appreciate the thoughtful commit names you two used to communicate.

**Pull request closed without merging**